frontend/components/common/CompatibleDevice.tsx

```diff
@@ -11,12 +11,17 @@ export function CompatibleDevice({
    device,
    truncate = 0,
 }: CompatibleDeviceProps) {
-   let variants = device.variants;
-   let unlistedVariants = 0;
-   if (truncate > 0 && variants.length > truncate) {
-      variants = device.variants.slice(0, truncate - 1);
-      unlistedVariants = device.variants.length - variants.length;
-   }
+   const variants = React.useMemo(
+      () => device.variants.filter(Boolean),
+      [device.variants]
+   );
+   const [visibleVariants, hiddenVariantCount] = React.useMemo(
+      () =>
+         truncate > 0 && variants.length > truncate
+            ? [variants.slice(0, truncate), variants.length - truncate]
```

Comment:
I personally still prefer the `truncate - 1` as `truncate` then refers the the total number of lines in a compatibility section, not how many variants can be shown.

Reply:
It isn't clear to me how "the total number of lines in a compatibility section" is different from the number of variants to show, could you elaborate? Does the total number of lines include the "and X more models..."?

I made this change because it seemed like the one caller was passing in a value that represented "the number of variants to show".

https://github.com/iFixit/react-commerce/blob/26b01d5123b21df836e4ad9bfabe92096a48c098/frontend/templates/product/sections/ProductSection/CompatibleDevices.tsx#L35

Reply:
I'm now realizing this is also one of those cases where instead of showing "and 1 more model..." we should just show the last model.

Reply:
Ok so i think what happens is in this current code, if truncate = 5 and there are 5 variants it will show all 5. If there are 6 it will show 5 variants and a line with a link to the extra 1 in the later compatibility section for a total of 6 lines. In the old version the we would if there were 5 we would show 5, but if there were 6 we would show 4 variants and 1 link line for a total of 5 lines. I just think thats cleaner, but I would also agree that its borderline arbitrary.